emcc

```diff
@@ -1491,6 +1491,13 @@ try:
       js_optimizer_queue = []
     js_optimizer_extra_info = {}
 
+  # Regardless of what -OX mode we're in, dead functions should be removed. This pass
+  # should be considered first so other optimisations aren't taken along for the ride.
+  if shared.Settings.DEAD_FUNCTIONS:
+    js_optimizer_queue += ['eliminateDeadFuncs']
+    js_optimizer_extra_info['dead_functions'] = shared.Settings.DEAD_FUNCTIONS
+    flush_js_optimizer_queue()
```

Comment:
no need to flush the queue here, when `js_opts` are on it should be flushed later anyhow